src/zenml/integrations/gcp/services/vertex_deployment.py

```diff
@@ -0,0 +1,431 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the Vertex AI Deployment service."""
+
+import re
+import time
+from typing import Any, Dict, Generator, List, Optional, Tuple
+
+from google.api_core import exceptions
+from google.cloud import aiplatform
+from google.cloud import logging as vertex_logging
+from pydantic import BaseModel, Field
+
+from zenml.client import Client
+from zenml.integrations.gcp.flavors.vertex_model_deployer_flavor import (
+    VertexBaseConfig,
+)
+from zenml.logger import get_logger
+from zenml.services import ServiceState, ServiceStatus, ServiceType
+from zenml.services.service import BaseDeploymentService, ServiceConfig
+
+logger = get_logger(__name__)
+
+# Increase timeout for long-running operations
+POLLING_TIMEOUT = (
+    1800  # Increased from 1200 to allow for longer deployment times
+)
+UUID_SLICE_LENGTH: int = 8
+
+
+def sanitize_vertex_label(value: str) -> str:
+    """Sanitize a label value to comply with Vertex AI requirements.
+
+    Args:
+        value: The label value to sanitize
+
+    Returns:
+        Sanitized label value
+    """
+    # Handle empty string
+    if not value:
+        return ""
+
+    # Convert to lowercase
+    value = value.lower()
+    # Replace any character that's not lowercase letter, number, dash or underscore
+    value = re.sub(r"[^a-z0-9\-_]", "-", value)
+    # Ensure it starts with a letter/number by prepending 'x' if needed
+    if not value[0].isalnum():
+        value = f"x{value}"
+    # Truncate to 63 chars to stay under limit
+    return value[:63]
+
+
+class VertexDeploymentConfig(VertexBaseConfig, ServiceConfig):
```

Comment:
You have some conflicting information in this class:

`model_id` - comes from `VertexBaseConfig`
`model_name` and `model_version` - come from `ServiceConfig`

You should get rid of `model_id` and just use `model_name` and `model_version`, as do all other model deployers.